src/devices/esp8266.c

```diff
@@ -329,6 +330,9 @@ bool esp8266_init(struct Serial *s, const size_t max_cmd_len,
         if (!_setup(s, max_cmd_len))
                 return false;
 
+        /* Hard reset our ESP8266*/
```

Comment:
Probably shouldn't put this here since MK2 would still have issues if this is a re-init scenario.  I would keep the method separate and I will make use of it when I complete #583